release.nix

```diff
@@ -178,15 +167,36 @@ let
         pkgsNew.pythonPackages.protobuf
         (oldAttrs : {doCheck = false; doInstallCheck = false;});
 
+    shell-hook-common =
+      pkgsNew.lib.optionalString pkgsNew.stdenv.isDarwin ''
+        export DYLD_LIBRARY_PATH=${pkgsNew.grpc}/lib''${DYLD_LIBRARY_PATH:+:}$DYLD_LIBRARY_PATH
+      '' +
+      pkgsNew.lib.optionalString pkgsNew.stdenv.isLinux ''
+        export LD_LIBRARY_PATH=${pkgsNew.grpc}/lib''${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH
+      '';
+
     test-grpc-haskell =
       pkgsNew.mkShell {
         nativeBuildInputs = [
-          (pkgsNew.haskellPackages.ghcWithPackages (pkgs: [
-                pkgs.grpc-haskell
-              ]
-            )
-          )
+          (with pkgsNew.haskellPackages; [
+             cabal-install
+             (ghcWithPackages (pkgs: with pkgs; [
+                grpc-haskell-core
+                # No need to guard nix-shell entry on passing package tests
+                (pkgsNew.haskell.lib.dontCheck grpc-haskell)
+                language-c
+                # Additional packages we need to have available for building
+                # and running tests in the nix-shell environment
+                mmorph pipes tasty tasty-hunit tasty-quickcheck
+              ]))
+           ])
+          (pkgsNew.python.withPackages (pkgs: [ pkgs.grpcio-tools ]))
         ];
+        shellHook = ''
+          ${shell-hook-common}
+          echo Running: cabal configure --extra-include-dirs ${pkgsNew.grpc}/include --extra-lib-dirs ${pkgsNew.grpc}/lib --enable-tests
+          cabal configure --extra-include-dirs ${pkgsNew.grpc}/include --extra-lib-dirs ${pkgsNew.grpc}/lib --enable-tests
```

Comment:
Something seems off if we have to explicitly specify the `include`/`lib` paths for `grpc`.  Normally I'd expect Nix to automatically handle that.  Did you try adding `grpc` as a `buildInput` for the shell?

Reply:
Yes, I tried a few different variations (including the one you mentioned) and continued to run into the same issue as described above. _However_, when I went to reproduce the problematic behavior, I'm not able to do so at the branch HEAD now. I suspect I can find the smoking gun and simplify this a lot (possibly even dropping the `test-grpc-haskell` attribute altogether).